ppb/sprites.py

```diff
@@ -125,12 +124,40 @@ def center(self):
 
     @center.setter
     def center(self, value):
-        if self.side in (TOP, BOTTOM):
-            setattr(self.parent, self.side, value[1])
-            self.parent.center.x = value[0]
-        else:
-            setattr(self.parent, self.side, value[0])
-            self.parent.position.y = value[1]
+        self.parent.position = self._mk_update_vector_center(value)
+
+    def _mk_update_vector_side(self, attribute, value):
+        value = Vector(value)
+        assert attribute != 'center'
+        # Does a bunch of dynamc resolution:
+        # Sprite.top.left
+        #        ^   ^ attribute
+        #        self.side
+        self_dimension, self_offset = self._lookup_side(self.side)
+
+        attr_dimension, attr_offset = self._lookup_side(attribute)
+
+        assert self_dimension != attr_dimension
+
+        fields = {
+            self_dimension: value[self_dimension] - self_offset,
+            attr_dimension: value[attr_dimension] - attr_offset,
+        }
+        return Vector(**fields)
```

Comment:
Is this correct?  I don't think `Vector()` would accept keyword arguments `attr_dimension` and `self_dimension`

Reply:
no, they're equal to `'x'` and `'y'`.